src/uproot/_util.py

```diff
@@ -313,7 +313,7 @@ def file_object_path_split(urlpath: str) -> tuple[str, str | None]:
 
     separator = "::"
     parts = urlpath.split(separator)
-    object_regex = re.compile(r"(.+\.root):(.*$)", re.IGNORECASE)
+    object_regex = re.compile(r"(^.*):(.*$)", re.IGNORECASE)
```

Comment:
```suggestion
    object_regex = re.compile(r"(.+\.root.*):(.*$)", re.IGNORECASE)
```
I think we can probably just use this regex to make it as generic as possible